### PR TITLE
[BugFix] Fix pending queryeis when group.concurrency_limit is null

### PR DESCRIPTION
**Pull request by ZiheLiu**
Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/scheduler/slot/SlotRequestQueue.java [hunk 1]
```diff
@@ -142,8 +142,8 @@ private boolean isGroupSlotAvailable(ResourceGroup group, int numAllocatedSlotsO
             return true;
         }
 
-        int numTotalSlots = group.getConcurrencyLimit();
-        return numTotalSlots <= 0 || numAllocatedSlotsOfGroup < numTotalSlots;
+        Integer numTotalSlots = group.getConcurrencyLimit();
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/scheduler/slot/SlotRequestQueue.java:145]: I think the default value for concurrencyLimit should be 0 or -1, means no limit. shouldn't return null for getConcurrencyLimit.

Comment (ZiheLiu) [reply fe/fe-core/src/main/java/com/starrocks/qe/scheduler/slot/SlotRequestQueue.java:145]: Yes, I agree with you. Use `Integer` could expose more bug ricks.
However, it is the logic long long ago. Maybe there is compatibility problem if I change it from `Integer` to `int`.

**Review by satanson — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 0 / 2 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/scheduler/slot/SlotRequestQueue.java|0|2|00.00%|[145, 146]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Pull request merged**